Source/WebKit/WebKitSwift/RealityKit/RKEntity.swift

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2024 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+// BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+
+#if os(visionOS)
+
+import Foundation
+import RealityKit
+import WebKitSwift
+import os
+import simd
+
+private extension Logger {
+    static let realityKitEntity = Logger(subsystem: "com.apple.WebKit", category: "RealityKitEntity")
+}
+
+@objc(WKSRKEntity)
+public final class WKSRKEntity: NSObject {
+    let entity: Entity
+
+    @objc(initWithCoreEntity:) init(with coreEntity: REEntityRef) {
+        entity = Entity.__fromCore(__EntityRef.__fromCore(coreEntity))
+    }
+
+    @objc(boundingBoxExtents) public var boundingBoxExtents: simd_float3 {
```

Comment:
Do we want to return a nullable here? (if it's something that can be supported when bridging into ObjC?)
Same for boundingBoxCenter

Reply:
Why do you think this needs to be nullable?

Reply:
Just because the underlying boundingBox object is nullable.

Reply:
I see - yeah, I think if I want to this to be able to return null, I'd probably have to return a `simd_float3*`, and if I do that, then the caller would have to handle the null result somehow. Maybe simpler to keep it the way it is.